media/plg_installer_webinstaller/js/client.es6.js

```diff
@@ -0,0 +1,359 @@
+      }
+
+      this.loadweb(`${webInstallerOptions.options.base_url}index.php?format=json&option=com_apps${tail}`);
+    }
+
+    static clicker() {
+      if (document.querySelector('.grid-view')) {
+        document.querySelector('.grid-view').addEventListener('click', () => {
+          webInstallerOptions.list = 0;
+          document.querySelector('.list-container').classList.add('hidden');
+          document.querySelector('.grid-container').classList.remove('hidden');
+          document.getElementById('btn-list-view').classList.remove('active');
+          document.getElementById('btn-grid-view').classList.remove('active');
+        });
+      }
+
+      if (document.querySelector('.list-view')) {
+        document.querySelector('.list-view').addEventListener('click', () => {
+          webInstallerOptions.list = 1;
+          document.querySelector('.grid-container').classList.add('hidden');
+          document.querySelector('.list-container').classList.remove('hidden');
+          document.getElementById('btn-grid-view').classList.remove('active');
+          document.getElementById('btn-list-view').classList.add('active');
+        });
+      }
+    }
+
+    /**
+     * @param {string} installUrl
+     * @param {string} name
+     * @returns {boolean}
+     * @todo Migrate this function's alert to a CE dialog
+     * @todo Migrate this function's hardcoded English string to Joomla.JText
+     */
+    static installfromweb(installUrl, name) {
+      if (!installUrl) {
+        alert("This extension cannot be installed via the web. Please visit the developer's website to purchase/download.");
+
+        return false;
+      }
+
+      const installUrlField = document.getElementById('install_url');
+      const uploadUrlContainer = document.getElementById('uploadform-web-url');
+
+      installUrlField.value = installUrl;
+      uploadUrlContainer.innerHTML = installUrl;
+
+      if (name) {
+        const nameElement = document.getElementById('uploadform-web-name');
+        nameElement.innerHTML = name;
+        document.getElementById('uploadform-web-name-label').classList.remove('hidden');
+      } else {
+        document.getElementById('uploadform-web-name-label').classList.add('hidden');
+      }
+
+      // jQuery('#jed-container').slideUp(300);
+      document.getElementById('uploadform-web').classList.remove('hidden');
+
+      return true;
+    }
+
+    /**
+     * Onclick handler for the button#appssubmitbutton element
+     *
+     * @param {string} redirectUrl
+     * @returns {boolean}
+     * @todo Convert from inline onclick registration, requires coordinated PR to IFW repo
+     * @todo Migrate this function's confirm to a CE dialog
+     * @todo Migrate this function's hardcoded English string to Joomla.JText
+     */
+    static installfromwebexternal(redirectUrl) {
+      const redirectConfirm = confirm(`You will be redirected to the following link to complete the registration/purchase - \n${redirectUrl}`);
```

Comment:
@mbabker try using `window.confirm` instead

Reply:
grr... this will work, but thats only a trick to fool the linter when you pass the window object as a local param. It will make the dog happy, but will not solve the problem.

Reply:
When we do it like this, it would make more sense to disable the rule. Unfortunately i do not have a perfect solution atm. We need some kind of notification layer (which should be more powerfull than renderMessages).

Reply:
`confirm` is a window object (see MDN: [Window.confirm()](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm)).

Reply:
We need to create our own UI element for this using custom elements, so ignore this for now